plugin.audio.criminalpodcast/resources/lib/criminalpodcast.py

```diff
@@ -0,0 +1,262 @@
+import requests
+import re
+from bs4 import BeautifulSoup
+
+def get_soup(url):
+    """
+    @param: url of site to be scraped
+    """
+    page = requests.get(url)
+    soup = BeautifulSoup(page.text, 'html.parser')
+    
+    print "type: ", type(soup)
```

Comment:
What would you suggest instead of print? Return?

Reply:
`xbmc.log(...)`

Reply:
I have checked the log at debug level, there's nothing in there that gives a warning about the print function:
[](https://pastebin.com/X9UAbPbD)

Reply:
https://codedocs.xyz/xbmc/xbmc/group__python__xbmc.html#ga3cd3eff2195727144ad6fbabd4744961

```
import xbmc
xbmc.log('type = %s'%(type(soup)),xbmc.LOGDEBUG)
```

Reply:
Many thanks, I have added that to the start of my python script (https://github.com/leopheard/criminalpodcast/blob/master/addon.py) but it is giving the following error?:

> ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
> - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
> Error Type: <type 'exceptions.SyntaxError'>
> Error Contents: invalid syntax (addon.py, line 2)
> File "/home/osmc/.kodi/addons/plugin.audio.criminalpodcast/addon.py", line 2
> xbmc.log('type: %s'%(type(soup)),xbmc.LOGDEBUG)lugin = Plugin()
>                                                                                                    ^
> SyntaxError: invalid syntax
> ->End of Python script error report<--

Reply:
The variable you are using (`soup`) has to be declared before it can be used. This is nothing special about `xbmc.log(...)`.

Reply:
I've moved it and now there's no error so I'm assuming it's good. Now there's the issue of my code referencing modules that are no longer there as xmbcswift2 isn't good anymore - could you point me towards any examples of a fully python3 / Kodi 19 compliant audio addon please? :

> ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
>- NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
>Error Type: <type 'exceptions.ImportError'>
>Error Contents: cannot import name Plugin
>Traceback (most recent call last):
>File "/.../plugin.audio.criminalpodcast/addon.py", line 2, in <module>
>from xbmc import Plugin, xbmcgui
>ImportError: cannot import name Plugin
>-->End of Python script error report<--